pkg/backoff/exponential.go

```diff
@@ -38,22 +40,28 @@ type ExponentialBackOff struct {
 }
 
 // NewExponentialBackOff creates an istio wrapped ExponentialBackOff.
+// By default, it never stops.
 func NewExponentialBackOff(initFuncs ...func(off *ExponentialBackOff)) BackOff {
-	b := &ExponentialBackOff{}
+	b := ExponentialBackOff{}
 	b.ExponentialBackOff = backoff.NewExponentialBackOff()
+	b.ExponentialBackOff.MaxElapsedTime = 0
 	for _, fn := range initFuncs {
-		fn(b)
+		fn(&b)
 	}
 	b.Reset()
-	return b
+	return &b
```

Comment:
I thought we had decided to do

```suggestion
	return b
```

or are there issues there?